flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogFunctionImpl.java

```diff
@@ -20,40 +20,36 @@
 
 import org.apache.flink.util.StringUtils;
 
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Optional;
 
 import static org.apache.flink.util.Preconditions.checkArgument;
-import static org.apache.flink.util.Preconditions.checkNotNull;
 
 /**
  * A catalog function implementation.
  */
 public class CatalogFunctionImpl implements CatalogFunction {
 	private final String className; // Fully qualified class name of the function
-	private final Map<String, String> properties;
+	private final boolean isGeneric;
 
-	public CatalogFunctionImpl(String className, Map<String, String> properties) {
+	public CatalogFunctionImpl(String className) {
+		this(className, true);
+	}
+
+	public CatalogFunctionImpl(String className, boolean isGeneric) {
 		checkArgument(!StringUtils.isNullOrWhitespaceOnly(className), "className cannot be null or empty");
 
 		this.className = className;
-		this.properties = checkNotNull(properties, "properties cannot be null");
+		this.isGeneric = isGeneric;
```

Comment:
felt like `isGeneric` can be automatically determined by checking if the class of `className` is assignable from Flink's udf interfaces? or it can even take a class as constructor param

Reply:
Good suggestion. I think checking whether the class is assignable from Flink's udf interfaces is more generic way to handle with it.

Reply:
I find actually there several test cases, such as GenericInMemoryCatalogTest use a raw class String without real class definition. Using the assignable class check either in CatalogFunctionImpl or test case will cause an extra code change. To limit the scope of this PR, How about we keep it as now for the consistency with how we handle generic lable of  table and view in hive catalog?

Reply:
we can change the test

Reply:
I feel the change won't be substantial and make the change all together would be better